lib/_http_server.js

```diff
@@ -349,7 +355,11 @@ function connectionListenerInternal(server, socket) {
   socket.on('timeout', socketOnTimeout);
 
   var parser = parsers.alloc();
-  parser.reinitialize(HTTPParser.REQUEST, parser[is_reused_symbol]);
+
+  parser.initialize(
+    HTTPParser.REQUEST,
+    new HTTPServerAsyncResource('HTTPINCOMINGMESSAGE', socket)
+  );
```

Comment:
I would add a comment above this call explaining why we are creating a new resource every time and why it's ok to do it here (parser events are not exposed to the end user).